src/patternfly/layouts/Toolbar/styles.scss

```diff
@@ -34,14 +45,24 @@
 }
 
 // Group
-.pf-l-toolbar__group:not(:last-child) {
```

Comment:
this introduced `margin-right` on the last toolbar group. that needs to be removed

Reply:
- added a new variable to the top list, `--pf-l-toolbar__group--MarginRight-Zero: var(0);`
- added `& > .pf-l-toolbar__group:last-child {margin-right: var(--pf-l-toolbar__group--MarginRight-Zero);}` in order to remove margin-right from last child.